### PR TITLE
increase severity of multiple errorprone checks

### PR DESCRIPTION
**Pull request by XN137**
note that we fix DifferentNameButSame with a minimal diff, but
not make it uniform across all files

### CODE REVIEW

**Comment by snazy** [comment]
Running `./gradlew compileAll` locally could be more efficient than force-pushing?

File: compatibility/compatibility-tests/src/intTest/java/org/projectnessie/tools/compatibility/tests/ITUpgradePath.java [hunk 1]
```diff
@@ -221,7 +219,7 @@ void commitLog() throws NessieNotFoundException {
               String commitMessage = "hello world " + versionFromRef;
               assertThat(commitLog)
                   .hasSize(1)
-                  .map(LogEntry::getCommitMeta)
+                  .map(LogResponse.LogEntry::getCommitMeta)
```
Comment (dimas-b) [inline compatibility/compatibility-tests/src/intTest/java/org/projectnessie/tools/compatibility/tests/ITUpgradePath.java:222]: why not keep the import for `LogEntry`?

File: gc/gc-iceberg-mock/src/main/java/org/projectnessie/gc/iceberg/mocks/MockManifestEntry.java [hunk 2]
```diff
@@ -51,7 +50,7 @@ public Status status() {
 
   public abstract String filePath();
 
-  public abstract StructType partitionType();
+  public abstract Types.StructType partitionType();
```
Comment (dimas-b) [inline gc/gc-iceberg-mock/src/main/java/org/projectnessie/gc/iceberg/mocks/MockManifestEntry.java:53]: Why do we sometimes import the inner class name, and sometime qualify it with its parent class name?

In this particular case, the inner class name seems to be non-ambiguous, so why do we change it to be qualified by the parent type?

Comment (XN137) [reply gc/gc-iceberg-mock/src/main/java/org/projectnessie/gc/iceberg/mocks/MockManifestEntry.java:53]: as stated in the PR description we resolve `DifferentNameButSame` violations by doing the minimal amount of changes... so this file uses more `Types.StructType` than `StructType` right now, we we prefer to use the former for the fix.

if we can come up with a uniform convention on how to handle these things, we can also apply that kind of fix.

imo the first step is making things uniform on a per file basis, wdyt?

Comment (dimas-b) [reply gc/gc-iceberg-mock/src/main/java/org/projectnessie/gc/iceberg/mocks/MockManifestEntry.java:53]: IMHO, shorter type references (i.e. importing the most specific name possible)  are preferable over smaller PR diff.

I'm open to other opinions too.

File: integrations/spark-extensions-basetests/src/main/java/org/projectnessie/spark/extensions/SparkSqlTestBase.java [hunk 3]
```diff
@@ -299,14 +298,13 @@ protected List<SparkCommitLogEntry> commitAndReturnLog(String branch, String ini
         content != null
             ? ImmutableOperations.builder()
                 .addOperations(
-                    Operation.Put.of(
-                        key, IcebergTable.of("foo", 42, 42, 42, 42, content.getId()), content))
+                    Put.of(key, IcebergTable.of("foo", 42, 42, 42, 42, content.getId()), content))
```
Comment (XN137) [inline integrations/spark-extensions-basetests/src/main/java/org/projectnessie/spark/extensions/SparkSqlTestBase.java:301]: is `Put` too short/common like `Builder` or not?

File: versioned/persist/serialize/src/main/java/org/projectnessie/versioned/persist/adapter/serialize/ProtoSerialization.java [hunk 4]
```diff
@@ -34,14 +34,13 @@
 import org.projectnessie.versioned.persist.adapter.RefLog;
 import org.projectnessie.versioned.persist.adapter.RepoDescription;
 import org.projectnessie.versioned.persist.serialize.AdapterTypes;
```
[the remark this answers is not in the document]
Comment (dimas-b) [reply versioned/persist/serialize/src/main/java/org/projectnessie/versioned/persist/adapter/serialize/ProtoSerialization.java:36]: SGTM

File: servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/tests/BaseTestNessieApi.java [hunk 5]
```diff
@@ -1440,7 +1439,8 @@ public void fetchEntriesByNamelessReference() throws BaseNessieClientServerExcep
             .operation(Put.of(b, tb))
             .commitMeta(CommitMeta.fromMessage("commit 1"))
             .commit();
-    List<Entry> entries = api().getEntries().hashOnRef(branch.getHash()).get().getEntries();
+    List<EntriesResponse.Entry> entries =
```
[the remark this answers is not in the document]
Comment (dimas-b) [reply servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/tests/BaseTestNessieApi.java:1442]: In the context of Nessie API (implied by this class), I think `Entry` has a specific meaning. I'd prefer keeping the import for it.

[the remark this answers is not in the document]
Comment (snazy) [reply servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/tests/BaseTestNessieApi.java:1442]: As mentioned, errorprone's `BadImport` rule applies.

File: versioned/persist/tx/src/main/java/org/projectnessie/versioned/persist/tx/TxDatabaseAdapter.java [hunk 6]
```diff
@@ -1673,7 +1672,7 @@ private RefLogEntry writeRefLogEntry(
       NamedRef ref,
       RefLogHead refLogHead,
       Hash commitHash,
-      Operation operation,
+      RefLogEntry.Operation operation,
```
Comment (XN137) [inline versioned/persist/tx/src/main/java/org/projectnessie/versioned/persist/tx/TxDatabaseAdapter.java:1675]: should we prefer `Operation` in this file instead?

Comment (dimas-b) [reply versioned/persist/tx/src/main/java/org/projectnessie/versioned/persist/tx/TxDatabaseAdapter.java:1675]: I think in this particular case using the longer name is justified to identify this as belonging to the deprecated ref log functionality.

**Review by dimas-b — APPROVED** [review]
LGTM, if @snazy is ok with these changes. Thanks for you effort, @XN137! :)

**Review by snazy — APPROVED** [review]

**Pull request merged**